src/Shared/CertificateGeneration/ProcessRunner.cs

```diff
@@ -0,0 +1,106 @@
+            {
+                process.StartInfo.FileName = "sudo";
+                if (!options.IsInteractive)
+                {
+                    process.StartInfo.ArgumentList.Add("--non-interactive");
+                }
+            }
+            else
+            {
+                process.StartInfo.FileName = options.Command[i++];
+            }
+            for (; i < options.Command.Count; i++)
+            {
+                process.StartInfo.ArgumentList.Add(options.Command[i]);
+            }
+            string commandLine = $"{process.StartInfo.FileName} {string.Join(" ", process.StartInfo.ArgumentList)}";
+            bool readStdErr = options.ThrowOnFailure || options.ReadStandardError;
+            StringBuilder? stdErr = null;
+            if (readStdErr)
+            {
+                stdErr = new StringBuilder();
+                process.ErrorDataReceived += (o, e) => {
+                    if (e.Data != null)
+                    {
+                        stdErr.AppendLine(e.Data);
+                    }
+                };
+            }
+            StringBuilder? stdOut = null;
+            bool readStdOut = options.ReadStandardOutput;
+            if (readStdOut)
+            {
+                stdOut = new StringBuilder();
+                process.OutputDataReceived += (o, e) => {
+                    if (e.Data != null)
+                    {
+                        stdOut.AppendLine(e.Data);
+                    }
+                };
+            }
+            process.Start();
+            process.BeginErrorReadLine();
+            process.BeginOutputReadLine();
+            if (!string.IsNullOrEmpty(options.StandardInput))
+            {
+                process.StandardInput.Write(options.StandardInput);
+            }
+            process.StandardInput.Close();
+            process.WaitForExit();
+            if (options.ThrowOnFailure && process.ExitCode != 0)
+            {
+                throw new Exception($"Command '{commandLine}' failed with {process.ExitCode}: {stdErr}.");
+            }
+            return new ProcessRunResult
+            {
+                ExitCode = process.ExitCode,
+                CommandLine = commandLine,
+                StandardError = stdErr,
+                StandardOutput = stdOut
+            };
+        }
+
+        public static bool HasProgram(string program)
+        {
+            string path;
+            string? pathEnvVar = Environment.GetEnvironmentVariable("PATH");
+            if (pathEnvVar != null)
+            {
+                string[] pathItems = pathEnvVar.Split(':', StringSplitOptions.RemoveEmptyEntries);
+                foreach (var pathItem in pathItems)
+                {
+                    path = Path.Combine(pathItem, program);
+                    if (File.Exists(path))
+                    {
+                        return true;
+                    }
+                }
+            }
+            return false;
+        }
```

Comment:
We should not do this. I think its best to rely on `which` to see if a tool is available.

Reply:
We're doing the same thing as .NET does when it looks at `PATH` to know what file `Process` should execute: https://github.com/dotnet/runtime/blob/8709d595f7105cbf73774ec762266bc56f99cf04/src/libraries/System.Diagnostics.Process/src/System/Diagnostics/Process.Unix.cs#L736-L754.

I've encountered minimal OS installs and container images where `which` was not installed.